sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -412,7 +412,7 @@ case class GpuHashAggregateExec(
             val defaultValues =
               aggregateFunctions.flatMap(_.initialValues)
             val vecs = defaultValues.safeMap { ref =>
-              withResource(GpuScalar.from(ref.asInstanceOf[GpuLiteral].value, ref.dataType)) {
+              withResource(ref.columnarEval(null).asInstanceOf[GpuScalar]) {
```

Comment:
We should never use `columnarEval(null)`.  This is only safe if we know exactly what the instance is, and if we know what the instance is then we should just call the direct method on that instance we need rather than the generic `columnarEval` interface with an invalid input.

Reply:
Got it